lib/_stream_readable.js

```diff
@@ -447,7 +447,8 @@ function maybeReadMore_(stream, state) {
 // for virtual (non-string, non-buffer) streams, "length" is somewhat
 // arbitrary, and perhaps not very meaningful.
 Readable.prototype._read = function(n) {
-  this.emit('error', new Error('not implemented'));
+  const err = new Error('not implemented');
+  process.nextTick(() => this.emit('error', err));
 };
```

Comment:
so, I would rewrite all of those as

```js
function myFunc () {
  let err = new Error('myerror')
  nextTick(emitError, this, err) 
  return
}

function emitError (stream, err) {
  stream.emit('error', err)
}
```

This should be the best way to handle this. But run the benchmarks before and after.